lib/bn.js

```diff
@@ -52,6 +52,7 @@ else
 
 BN.BN = BN;
 BN.wordSize = 26;
+BN._hasBuffer = typeof Buffer !== 'undefined';
```

Comment:
I'm not sure if it will work well when `bn.js` is used as dependency. Usually, I work-around such problems by doing `require('buf' + 'fer').Buffer` inside of `try { } catch {}`.

In this case, though, such check should happen lazily, IMO.